### PR TITLE
New filter framework

### LINKED ISSUE #6: Filter framework
Build out framework to support adding, and applying image filters to ASCII grids

### PR DESCRIPTION
**Pull request by Davidy22**
Two sample filters to demonstrate usage and various fixes also included. Resolves #6

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -56,3 +60,7 @@
         # sys.exit(0)
     except ResizeScreenError as e:
         last_scene = e.scene
+    except StopApplication:
+        quit(0)
+    except KeyboardInterrupt:
+        quit(0)
```
Comment (dhananjaylatkar) [inline src/main.py:66]: should close the screen before quiting

File: src/filters/snowman.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from typing import List, Tuple
+
+
+class filter:
+    """Sample filter for ASCII operations"""
+
+    def __init__(self):
```
Comment (garuna-m6) [inline src/filters/snowman.py:7]: you can inherit  a base filter class

Comment (Davidy22) [reply src/filters/snowman.py:7]: We can't, filters are dynamically loaded from a context where they can't directly access the main code to modify internal values/instantiate from classes.

File: src/main.py (hunk 1 above)
Comment (Davidy22) [reply src/main.py:74]: Done 310f53f86db25b67229ea2ca6bf4086bad3da31e

**Pull request merged**